caddyhttp/proxy/upstream.go

```diff
@@ -416,9 +423,21 @@ func (u *staticUpstream) healthCheck() {
 			}
 
 			if r, err := u.HealthCheck.Client.Do(req); err == nil {
-				io.Copy(ioutil.Discard, r.Body)
-				r.Body.Close()
-				unhealthy = r.StatusCode < 200 || r.StatusCode >= 400
+				if u.HealthCheck.ContentString != "" {
+					b, err := ioutil.ReadAll(r.Body)
+					r.Body.Close()
+					if err != nil {
+						unhealthy = true
+					} else {
+						if !bytes.Contains(b, []byte(u.HealthCheck.ContentString)) {
+							unhealthy = true
+						}
+					}
+				} else {
+					io.Copy(ioutil.Discard, r.Body)
+					r.Body.Close()
+					unhealthy = r.StatusCode < 200 || r.StatusCode >= 400
```

Comment:
Is it intentional that we ignore the response status code when we check the contents of the body?